### PR TITLE
Update index with new LNF and modified layout

### PR DESCRIPTION
**Pull request by insectengine**

### CODE REVIEW

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 1]
```diff
@@ -35,140 +56,239 @@
...
...97272PdGvtSU+3HiHmf5EvcvzW0/blLkJY0v5ezH5fQvBXxv7PfmPgHYj6vzflxd9FV2mT4B6PdmvoT7cZiXeqYrwJdK9Xtv6Owex/1L7HuPS9+7BF8K+95HE1+ijKR9HzfhfB83lslLp6p8ie9xE8Jf8vKlOB9l+dIh0+/NfIn7vf3+0l6dL/E9jvkS3OPs8/KlLykufenypQcjvjQ+YPnSp+h7l+BLG9lfutV3j6MH97hAXnoT8pLiL1EM8vpLqe99h+BL4Hu/pN3j9qX+EgEmzktZf+nx1F96vLi/hHyJ/SXOS1ekvjfyJc5LJwJf+m17fI+72n+PY3/pCuwTiO9xFyS+dyXNS4Mtn+9Nb46+98WGL6l9As/HeemrQF8l7MfpfKnJ+3Ga7419AsZf6h8qvB/HfQLIl/J971sU39vfJ9AWfEn3l2pV+mU7/Uvd0vfG/Tj0vfvV/bglnJewTwD9pfB+XFXdjxN8Sb3H1bO+t94nAPtx6T2u39PvbfsE9LxUfD8O+wRK7sfl+94ND1/S+70nPX0CVfa9hyRfOkbvq8R+78v0/Th63v24Zdo9Dn3v8H7cMtyPa9Brd6G/tLw5/XBlLnxpqlO+hL63zUsb2F/y8qUpT14avkr0e2v3uJE4L1l/SfAl8Je0vHRqkXucfZSXwPc2ecntEzik86XdOl/aK/oEOC8J39vwpZVaXhqhPgGZl3J9b85L0vcO3OPoYzzyvTEv0RP3uIuAL0FeuhN8b+gTuPbOfN97B/Il8pd+jvIS+N47bL/3E7ueQH/Jx5f2MV/64w+KS+49Lv7OLeB7v8t5KZ8vRZCK+wTQ98a8xPe4c3k/jvnSbbHvfdd20SfAfMnkJegTMHxpifCXBgcHbb/36NrUX+rhvd2IL7W1/biA730Y7Mf9O30Cyyxfamb244gvreH9ONEnsNq/Hzeq78dNrvbuobREXgrux13s8qUW+EvF9+MWW76Evre+h3Kf6BN4jPsEyvrebZ/vLfuXmr79uBXJfhztoIi8JPfj+rz7cT3Il8D3VvbjoE8A+BL3L2X24xqiT4D4kn8/rqHzpfB+XMD3/hV8b5UvXWb5kr3HbYV7XJP7vYvxJXqK711wP25a7sdRXkLfm/6/1v0l4EsLg/txzJeadM38BzMdgVIu5rg8AAAAAElFTkSuQmCC) repeat-x, #0D1C2C;
```
Comment (gsmet) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:77]: Is this normal? It doesn't look correct.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 2]
```diff
@@ -35,140 +56,239 @@
...
-            </ul>
+    </div>
+</div>
+
+<div class="content-wrapper">
+    <div class="grid">
+        <div class="left-column">
+            <h1>You just made a Quarkus application.</h1>
+            <h4>Try running Quarkus Dev and <a href="/q/dev/">Dev UI</a> to view your installed extensions.</h4>
+            <p>This page is served by Quarkus. The source is in <code>src/main/resources/{config.file-name}</code>.</p>
```
Comment (gsmet) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:268]: The `config.file-name` variable doesn't look right for this item.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 3]
```diff
@@ -35,140 +56,239 @@
...
+            <p>Your static assets are located in <code>src/main/resources/META-INF/resources</code>.</p>
+            <p>Configure your application in <code>src/main/resources/{config.file-name}</code>.</p>
+            <p>Play with the provided code located in <code>{language.dir.code}</code></p>
```
Comment (gsmet) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:271]:
```suggestion
            <p>Play with the provided code located in <code>{language.dir.code}</code>.</p>
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I haven't looked at the result but I spotted a few things.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 1 above)
Comment (gsmet) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:77]: Hmmm, ignore this one, looks like there's a bug in the GitHub diff rendering.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 4]
```diff
@@ -4,27 +4,48 @@
...
...ObdiT5CVCMZ9wTRSgZ8sqTCCLgIUsl1NjXIkjwfG1JztBXt96a6QXM1IwXNQ0S07P8/0sE+3n6gSAJGkEnGH/3IQALr73H5wgnOMMFrnDjzoMnL95/meBLMwDbQBzf88NF8RdgkMGGGGqY4UYYaZTRxuAJxBckWIj8/8GMM94EEwmECRchUpRoQjFixYmXIFGSZClSpUmXIVOWbDly5Zlksimmmma6GURmmmW2OcTBxAYNGp202GdNurRbZZuNYYc2T9VbECzCHp2WaHHey3DAatv95le/W2+XKy7bTUKqm8w1clZX3XLdDTd9oXDPbXf0UPrRPA/d94DKN31azaWWT0tDZy29AgZGhcxMihT7qkSZUuUqVThinWpVatTq1e+YR/bY67jHXngSjoQT4Uy4EK6EW7iHR3iGV3iHT3wXvvbZ75DDLjjgoIua7YgBTjkd7BioI0xRIBg9EYJeUqWpKFQjtnFkCGVRqssJnzKkMIjvWbyOMMKYYgdH7mYwEAbSQBkRBtqINDBGlBEdxw8iwvMI678bs1Eu1an945CUAE1R/tsZcmVFcYeiEBwLBtJA3ZMI3L94VfGoyNBHpVG6/0/JPoVu3cG4olm0qsINlHaBlIJJTP6kNHNu5s1m7piS06tQ9z1UfTHKnoxiOCn50BeSDmlkd1pT2qmsFpWk1ZviNiRqIyHCxihBLZuZdSijVsGvMMSr+kivUsUtjckpFLvQmqwihZnXwsh5GpupmUpJyUxOFD1JU5ISHLTYpEYqMUSTEipy2JdCCvogBikEdhCicMhmglcI66Ui3hNUHBhSYj2KifGYFe2RBgb3i+xQEagQQoQHQQL8eVF/J5CbMAOFQOAwHDBb92eP2B0B8XWDQHzc9GFviAovCA56ebg6C6+uHhCwy/92Gef/ASHdIAInyhkj8C5idsF0w4o/uN8QAeSqiFPL37wT8nC+/qzcX79XnlV71SV/vzeBbNLoQp6uPykP1x+U2SVTXlQn1UH1v5kwBOF98978AxMwMxitPyp31++UR9VOdUnvdyYkywzzL+/jH/yr6/tRpUbTI4hO3gCHne0t8CviCvJ6p6EsbiQh/qq++/kT5jdH+KsnvzB4u8EUqJhUTM5CzY/NCrNi4h8z9nAJxBJ005kthDklKzud9Y8gwyaI2Hn7a3YR9exe1AAA) format('woff2'),
```
Comment (maxandersen) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:10]: Thats a big chunk of blob to have at beginning of file - will it make a difference for the browswer if we put this last so actual content of page is what we show user first ?

Comment (insectengine) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:10]: The CSS has to be in the head of the file for it to work. One of the parameters set by erin was that everything had to be in one file (images, fonts, etc) so there is no outside dependencies. The large blogs are Base64 encoded images (header background) and svg (logo) as well as the Open Sans font.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 2 above)
Comment (insectengine) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:268]: I was copying it from the previous page. I have no idea what it should be.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 1 above)
Comment (insectengine) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:77]: One of the parameters set by Erin was that everything had to be in one file (images, fonts, etc) so there is no outside dependencies. The large blobs are Base64 encoded images (header background) and svg (logo) as well as the Open Sans font.

**Review by gastaldi — DISMISSED** [review]
The new project creation fails with:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create (default-cli) on project standalone-pom: Failed to generate Quarkus project: Error while rendering template: /codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html: TemplateException: No namespace resolver found for [fill] in expression {fill:#fff;} in template /codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html on line 240 -> [Help 1]
```

**Comment by gastaldi** [comment]
Pushed a fix to [`1e7cb62` (#21233)](https://github.com/quarkusio/quarkus/pull/21233/commits/1e7cb624792b462b76fd69e3f72708d98979ca33)

**Comment by gastaldi** [comment]
Here is a preview from a generated project: 

image

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html [hunk 5]
```diff
@@ -35,142 +56,241 @@
...
+                <h3>These extensions were used in the creation of this project:</h3>
+<!-- Start restyle merged extensions content: {merged-content} --> 
+                <ul>
+                    <li>RESTEasy JAX-RS</li>
+                    <li>gRPC</li>
+                    <li>RESTEasy JAX_RS</li>
+                    <li>REASEasy Qute</li>
+                    <li>WebSockets</li>
+                </ul>
+<!-- End restyle merged extensions content: {merged-content} -->
```
Comment (gastaldi) [inline independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:283]: This was supposed to be replaced with the actual list of extensions, but apparently that's not happening

Comment (insectengine) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:283]: The is something we discussed with Erin that will require a change.

@ebullient  - How do we update the system to generate the the list of extensions?

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 4 above)
Comment (ebullient) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:10]: y. people aren't going to read the raw source of this file, Max. And if they do, it's pretty obvious what this is.. I don't have an issue here.

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 5 above)
Comment (ebullient) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:283]: You need this code from the previous version of the file: 
```
{merged-content}
```

Comment (insectengine) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html:283]: ok... y'all will then format the content into the UL?

**Comment by gastaldi** [comment]
I rebased and squashed the commits

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 5 above)
Comment (gastaldi) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html]: The `{merged-content}` doesn't seem to add the list of extensions here, but maybe I'm missing something. @ebullient can I ask you to give it a try?

**Comment by gsmet** [comment]
I force pushed an update with some minor improvements and hopefully fixing all the tests.

**Review by ebullient — APPROVED** [review]

File: independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html (hunk 5 above)
Comment (gsmet) [reply independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus/project/quarkus/base/src/main/resources/META-INF/resources/index.tpl.qute.html]: OK, so things are not perfect here but I'd like to discuss them with Andy when he's back. In the meantime, I will merge this.

**Review by gsmet — APPROVED** [review]

**Pull request merged**